Allow generating example request bodies for multipart/form-data

kylef opened the pull request:
Request example bodies are now generated for formData parameters while using the `multipart/form-data` consumes content type.

#### Dependencies

- [x] https://github.com/apiaryio/swagger-zoo/pull/47

pksunkara commented on src/parser.js:
```diff
@@ -53,6 +48,33 @@ function isJsonContentType(contentType) {
+function isMultiPartFormData(contentType) {
+  try {
+    const type = typer.parse(contentType);
+    return type.type === 'multipart' || type.subtype === 'form-data';
```
Shouldn't this be `&&`? Similarly below.

kylef replied:
Yes, I've fixed this. The problem exists in another method too which I fixed, I copied the similar code and adapted for multipart.

pksunkara commented on src/parser.js:
```diff
@@ -882,12 +904,21 @@ export default class Parser {
+        // add a default one `md5('boundary')` which hopefully isn't found
+        // in an example content
+        contentType += '; boundary=e4ef8847482d240d0a3ac612876ba8d1';
+      }
```
I think this boundary generation should be done in generator.js

kylef commented on src/generator.js:
```diff
@@ -22,9 +32,17 @@ export function bodyFromSchema(schema, payload, parser, contentType = 'applicati
-      } else if (contentType === 'multipart/form-data') {
-        // TODO: Unimplemented
-        return null;
+      } else if (type && type.type === 'multipart' && type.subtype == 'form-data') {
```
It was in separate file, so I've moved the logic into `media-type.js` which both files import and use now.

kylef commented:
I've updated this to do the following:

- Changed default boundary from `md5(boundary)` to just `BOUNDARY` as I think this behaviour is clearer than an obscure hash, however it does increase the risk that the boundary clashes with some example values or keys.
- Moved all media-type specific logic to separate file `media-type.js`

pksunkara commented:
Looks good. Didn't check why the tests are failing though.
